src/Model/GroundWaterFlow/gwf-ghba.f90

```diff
@@ -0,0 +1,467 @@
+      &NO DATA VALUE OR LESS THAN ZERO')"
+    character(len=*), parameter :: fmtconderr = &
+      "('GHBA BOUNDARY (',i0,') CONDUCTANCE (',g10.3,') IS NO DATA VALUE &
+      &OR LESS THAN ZERO')"
+    !
+    ! -- check stress period data
+    do i = 1, this%nbound
+      noder = this%nodelist(i)
+      if (noder == 0) cycle
+      bt = this%dis%bot(noder)
+      ! -- accumulate errors
+      if (this%bhead(i) < bt .and. this%icelltype(noder) /= 0) then
+        write (errmsg, fmt=fmtghberr) i, this%bhead(i), bt
+        call store_error(errmsg)
+      end if
+      if (this%iauxmultcol > 0) then
+        if (this%auxvar(this%iauxmultcol, i) == DNODATA .or. &
+            this%auxvar(this%iauxmultcol, i) < DZERO) then
+          write (errmsg, fmt=fmtcondmulterr) &
+            i, this%auxvar(this%iauxmultcol, i)
+          call store_error(errmsg)
+        end if
+      end if
+      if (this%cond(i) == DNODATA .or. &
+          this%cond(i) < DZERO) then
+        write (errmsg, fmt=fmtconderr) i, this%cond(i)
+        call store_error(errmsg)
+      end if
+    end do
+    !
+    !write summary of ghb package error messages
+    if (count_errors() > 0) then
+      call store_error_unit(this%inunit)
+    end if
+  end subroutine ghba_ck
+
+  !> @brief Formulate the HCOF and RHS terms
+  !!
+  !! Skip if no GHBs
+  !<
+  subroutine ghba_cf(this)
+    ! -- dummy
+    class(GhbaType) :: this
+    ! -- local
+    integer(I4B) :: i, noder
+    !
+    do i = 1, this%nbound
+      noder = this%nodelist(i)
+      if (noder == 0) cycle
+      if (this%ibound(noder) .le. 0) then
+        this%hcof(i) = DZERO
+        this%rhs(i) = DZERO
+        cycle
+      end if
+      this%hcof(i) = -this%cond_mult(i)
+      this%rhs(i) = -this%cond_mult(i) * this%bhead(i)
+    end do
+  end subroutine ghba_cf
+
+  !> @brief Copy rhs and hcof into solution rhs and amat
+  !<
+  subroutine ghba_fc(this, rhs, ia, idxglo, matrix_sln)
+    ! -- dummy
+    class(GhbaType) :: this
+    real(DP), dimension(:), intent(inout) :: rhs
+    integer(I4B), dimension(:), intent(in) :: ia
+    integer(I4B), dimension(:), intent(in) :: idxglo
+    class(MatrixBaseType), pointer :: matrix_sln
+    ! -- local
+    integer(I4B) :: i, noder, ipos
+    real(DP) :: cond, bhead, qghba
+    !
+    ! -- pakmvrobj fc
+    if (this%imover == 1) then
+      call this%pakmvrobj%fc()
+    end if
+
+    do i = 1, this%nbound
+      noder = this%nodelist(i)
+      if (noder == 0) cycle
+      rhs(noder) = rhs(noder) + this%rhs(i)
+      ipos = ia(noder)
+      call matrix_sln%add_value_pos(idxglo(ipos), this%hcof(i))
+      !
+      ! -- If mover is active and this boundary is discharging,
+      !    store available water (as positive value).
+      bhead = this%bhead(i)
+      if (this%imover == 1 .and. this%xnew(noder) > bhead) then
+        cond = this%cond_mult(i)
+        qghba = cond * (this%xnew(noder) - bhead)
+        call this%pakmvrobj%accumulate_qformvr(i, qghba)
```

Comment:
I think it is good that nbound is set to nodes user.  While we have to skip over inactive cells, it allows us easier output, if I remember correctly.  Is the `i` value that we pass in to `accumulate_qformvr` consistent with this?  It probably is, but just wanted to mention it.

Reply:
The pakmvrobj is initialized to maxbound (nodes user) for providers in BoundPackage so I think it is consistent.  I did look for a GHB test that used the mover that I could also update for GHBA but don't remember seeing one.  I'll look again- it would be nice to verify the GHBA mover is working as expected.

Reply:
Planning to add a test to https://github.com/MODFLOW-ORG/modflow6-testmodels